server/src/controllers/product/product.controller.ts

```diff
@@ -1,20 +1,26 @@
-import { ProductService } from "../../services/services";
-import { Request, Response } from "express-serve-static-core";
+import { productService } from "@/services/services";
+import { Request, Response, NextFunction} from "express";
+import {Product} from "@/db/entities/product.entity";
 
 export class ProductController {
-  constructor(private productService: ProductService) {
-    this.productService = productService;
-  }
 
-  public getAll = async (req: Request, res: Response) => {
-    const products = await this.productService.getAll();
-    res.json(products);
+  public getAll = async (req: Request, res: Response, next:NextFunction) => {
+    try{
+      res.json(await productService.getAll())
+    }catch (e){
+      next(e)
+    }
   }
 
-  public async getByProductTagId(req: Request, res: Response) {
-    const products = await this.productService.getByProductTagId(
-      parseInt(req.params.id)
-    );
-    res.json(products);
+  public async getByProductTagId(req: Request<{id}>, res: Response<Product[]>, next:NextFunction) {
+    try {
+      const products = await productService.getByProductTagId(
+          parseInt(req.params.id)
+      );
+      res.json(products);
```

Comment:
Also, when we send response, there should be specified status (in every controller response)